storage/src/main/java/org/apache/kafka/storage/internals/log/RemoteIndexCache.java

```diff
@@ -169,7 +169,16 @@ public Cache<Uuid, Entry> internalCache() {
     public void remove(Uuid key) {
         lock.readLock().lock();
         try {
-            internalCache.invalidate(key);
+            internalCache.asMap().computeIfPresent(key, (k, v) -> {
+                try {
+                    v.markForCleanup();
+                    v.cleanup();
```

Comment:
`markForCleanup` is enough, it'll set the cleanup flag as true, and rename files. `cleanup` can be done in a cleanup thread.

Reply:
Also, we have to add that entry to the `expiredIndexes` so that cleanup thread will be invoked on the removed entry.

Reply:
Done